apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/Internal/Logging/Logger.cs

```diff
@@ -26,32 +26,30 @@ namespace Google.Cloud.Spanner.V1.Internal.Logging
     /// </summary>
     public abstract class Logger
     {
-        private static Logger s_instance;
+        private static Logger s_defaultLogger = new DefaultLogger();
         private int _perfLoggingTaskEnabled;
         private readonly Dictionary<string, PerformanceTimeEntry> _perfCounterDictionary
             = new Dictionary<string, PerformanceTimeEntry>();
 
         /// <summary>
         /// This is an internal property and is not intended to be used by external code.
         /// </summary>
-        public static Logger DefaultLogger
-        {
-            get => s_instance ?? (s_instance = new DefaultLogger());
-            private set => s_instance = value;
-        }
+        public static Logger DefaultLogger => Interlocked.CompareExchange(ref s_defaultLogger, null, null);
 
         /// <summary>
-        /// This is an internal property and is not intended to be used by external code.
+        /// This is an internal method and is not intended to be used by external code.
         /// </summary>
+        /// <param name="instance">The new default logger. Must not be null.</param>
         public static void SetDefaultLogger(Logger instance)
         {
-            DefaultLogger = instance;
+            GaxPreconditions.CheckNotNull(instance, nameof(instance));
+            Interlocked.Exchange(ref s_defaultLogger, instance);
         }
 
         /// <summary>
         /// This is an internal property and is not intended to be used by external code.
         /// </summary>
-        public LogLevel LogLevel { get; set; } = LogLevel.Error;
+        public LogLevel LogLevel { get; set; } = LogLevel.None;
```

Comment:
Can't this change break client processes for clients who are logging errors without having to set this explicitly?
And, if this property is not supposed to be used by external code, who sets the logging level to something higher that `LogLevel.None`. I feel there's something I'm missing, but I don't know what.

Reply:
It's a breaking change effectively, but that's okay as we're at 2.0-betaXX. We'll include it in release notes. Fundamentally, I don't think we should ever be writing to the console by default.

I agree that the "is this meant to be used by external code or not" part is ambiguous at the moment - let's bear that in mind for future review.

Reply:
Ok